src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -422,12 +422,17 @@ Status GcsActorManager::RegisterActor(const ray::rpc::RegisterActorRequest &requ
   }
 
   actor_to_register_callbacks_[actor_id].emplace_back(std::move(success_callback));
-  RAY_CHECK(registered_actors_.emplace(actor->GetActorID(), actor).second);
+
+  // If GCS restarts before reply to GCS client and after writes to storage,
+  // GCS client will resend the `RegisterActor` request. After GCS restarts,
+  // `registered_actors_` and `unresolved_actors_` may recover data from storage, so we
+  // override them directly without checking.
+  registered_actors_.emplace(actor->GetActorID(), actor);
```

Comment:
Aren't we supposed to just return the method in this case?

Reply:
No, after GCS restarts, it is required to continue to create actor successfully.

Reply:
Can you write a comment?